src/XMLSecurityKey.php

```diff
@@ -267,183 +284,149 @@ public static function getRawThumbprint($cert)
-
-    private function encryptOpenSSL($data)
-    {
-        if ($this->cryptParams['type'] == 'public') {
-            if (! openssl_public_encrypt($data, $encrypted_data, $this->key, $this->cryptParams['padding'])) {
-                throw new Exception('Failure encrypting Data');
-            }
-        } else {
-            if (! openssl_private_encrypt($data, $encrypted_data, $this->key, $this->cryptParams['padding'])) {
-                throw new Exception('Failure encrypting Data');
-            }
-        }
-        return $encrypted_data;
-    }
-
-    private function decryptOpenSSL($data)
-    {
-        if ($this->cryptParams['type'] == 'public') {
-            if (! openssl_public_decrypt($data, $decrypted, $this->key, $this->cryptParams['padding'])) {
-                throw new Exception('Failure decrypting Data');
-            }
-        } else {
-            if (! openssl_private_decrypt($data, $decrypted, $this->key, $this->cryptParams['padding'])) {
-                throw new Exception('Failure decrypting Data');
-            }
-        }
-        return $decrypted;
-    }
-
-    private function signOpenSSL($data)
-    {
-        $algo = OPENSSL_ALGO_SHA1;
-        if (! empty($this->cryptParams['digest'])) {
-            $algo = $this->cryptParams['digest'];
-        }
-        if (! openssl_sign($data, $signature, $this->key, $algo)) {
-            throw new Exception('Failure Signing Data: ' . openssl_error_string() . ' - ' . $algo);
-        }
-        return $signature;
-    }
-
-    private function verifyOpenSSL($data, $signature)
+    /**
+     * @param $key
+     *
+     * @param bool|false $isFile
+     *
+     * @param bool|false $isCert
+     *
+     * @throws XMLSecLibsException
+     */
+    public function loadKey($key, $isFile = false, $isCert = false)
     {
-        $algo = OPENSSL_ALGO_SHA1;
-        if (! empty($this->cryptParams['digest'])) {
-            $algo = $this->cryptParams['digest'];
+        switch ($this->cryptParams['library']) {
+            case 'openssl':
+                $objExtension = new OpenSSL($this->cryptParams, $this->key);
+                $objExtension->setPassphrase($this->passphrase);
+                $objExtension->setType($this->type);
+                break;
+            case 'mcrypt':
+                $objExtension = new Mcrypt($this->cryptParams, $this->key);
+                $objExtension->setPassphrase($this->passphrase);
+                $objExtension->setType($this->type);
+                break;
+            default:
+                throw new XMLSecLibsException('Not implemented yet for this type.');
         }
-        return openssl_verify($data, $signature, $this->key, $algo);
+        return $objExtension->loadKey($key, $isFile, $isCert);
     }
 
+    /**
+     * @param $data
+     *
+     * @return mixed|string
+     *
+     * @throws XMLSecLibsException
+     */
     public function encryptData($data)
     {
         switch ($this->cryptParams['library']) {
```

Comment:
This switch can be refactored as a helper method. The code does not depends of the value $data